src/metabase/query_processor/pivot.clj

```diff
@@ -342,48 +347,52 @@
-                                                                              ;; if the col has a card-id, we swap the :lib/source to say source/card
-                                                                              ;; this allows `lib/find-matching-column` to properly match a column that has a join-alias
-                                                                              ;; but whose source is a model
-                                                                              (contains? col :lib/card-id) (assoc :lib/source :source/card))))
-                                                             (lib/breakouts-metadata mlv2-query))]
-                                 (fn [legacy-ref]
-                                   (try
-                                     (::i (lib.equality/find-column-for-legacy-ref
-                                           mlv2-query
-                                           -1
-                                           legacy-ref
-                                           breakouts))
-                                     (catch Throwable e
-                                       (log/errorf e "Error finding matching column for ref %s" (pr-str legacy-ref))
-                                       nil)))))
-
-        pivot-rows (when column-split-rows
-                     (into [] (keep index-in-breakouts) column-split-rows))
-        pivot-cols (when column-split-columns
-                     (into [] (keep index-in-breakouts) column-split-columns))]
-    {:pivot-rows pivot-rows
-     :pivot-cols pivot-cols}))
+  (let [column-split          (:pivot_table.column_split viz-settings)
+        column-split-rows     (seq (:rows column-split))
+        column-split-columns  (seq (:columns column-split))
+        column-split-measures (seq (:values column-split))
+        metadata-provider     (or (:lib/metadata query)
+                                  (lib.metadata.jvm/application-database-metadata-provider (:database query)))
+        mlv2-query            (lib/query metadata-provider query)
+        breakouts             (into []
+                                    (map-indexed (fn [i col]
+                                                   (cond-> col
+                                                     true                         (assoc ::i i)
+                                                     ;; if the col has a card-id, we swap the :lib/source to say source/card
+                                                     ;; this allows `lib/find-matching-column` to properly match a column that has a join-alias
+                                                     ;; but whose source is a model
+                                                     (contains? col :lib/card-id) (assoc :lib/source :source/card))))
+                                    (concat (lib/breakouts-metadata mlv2-query)
+                                            (lib/aggregations-metadata mlv2-query)))
```

Comment:
Key change here is adding the aggregations metadata to the list of checked cols. This is necessary since the :pivot-measures must be aggregations.